### PR TITLE
Add config file and socks5 support

### PR DESCRIPTION
**Pull request by kpcyrd**
Replaces curl with reqwest and introduces config files, the resolution order is `argument > ~/.config/arch-audit.conf > /etc/arch-audit.conf > default value`.

Configuring arch-audit to use tor looks like this, all settings are optional in the config file:
```
[network]
#source = "https://security.archlinux.org/all.json"
proxy = "socks5://127.0.0.1:9050"
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    pub proxy: Option<String>,
+}
+
+impl Config {
+    pub fn load(args: &Args) -> Result<Self> {
+        let mut merged = ConfigFile::default();
+
+        let mut configs = vec![PathBuf::from("/etc/arch-audit.conf")];
+        if let Some(dir) = dirs_next::config_dir() {
+            let path = dir.join("arch-audit.conf");
```
Comment (anthraxx) [inline src/config.rs:20]: I would prefer to namespace the configs into an `arch-audit` directory for both locations. maybe the file should also be `settings.conf` or `settings.toml` or similar, whose mime-type would work best?

**Comment by anthraxx** [comment]
btw i would have highly appreciated to get two atomic commits for "socks5 support" and "config file support". I mostly say if a commit message contains "and" its most likely not atomic enough :cat:

**Comment by kpcyrd** [comment]
@anthraxx done! :+1:

File: src/config.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    pub proxy: Option<String>,
+}
+
+impl Config {
+    pub fn load(args: &Args) -> Result<Self> {
+        let mut merged = ConfigFile::default();
+
+        let mut configs = vec![PathBuf::from("/etc/arch-audit/arch-audit.toml")];
+        if let Some(dir) = dirs_next::config_dir() {
+            let path = dir.join("arch-audit/arch-audit.toml");
```
Comment (anthraxx) [inline src/config.rs:20]: I'm confused, are we now using `settings.toml` inside the directory as the contrib example files is named or `arch-audit.toml`? :D i guess settings is better suited as arch-audit is already the context from within the namespace.

Comment (kpcyrd) [reply src/config.rs]: right, changed it back to settings.toml

**Comment by anthraxx** [comment]
applied in 58de1f9887a01ce41250a7597a610a8745472501
thanks a lot :smile_cat:

**Pull request closed without merging**